src/arc.rs

```diff
@@ -438,3 +442,39 @@ impl<T: Serialize> Serialize for Arc<T> {
         (**self).serialize(serializer)
     }
 }
+
+unsafe impl<T, U: ?Sized> unsize::CoerciblePtr<U> for Arc<T> {
```

Comment:
There should be a comment explaining why this `unsafe impl` is safe to perform in this case